Maple2.Server.Game/Packets/FishingPacket.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Numerics;
+using Maple2.Model.Common;
+using Maple2.Model.Enum;
+using Maple2.Model.Error;
+using Maple2.Model.Game;
+using Maple2.Model.Metadata;
+using Maple2.PacketLib.Tools;
+using Maple2.Server.Core.Constants;
+using Maple2.Server.Core.Packets;
+using Maple2.Tools.Extensions;
+
+namespace Maple2.Server.Game.Packets;
+
+public static class FishingPacket {
+    private enum Command : byte {
+        Prepare = 0,
+        Stop = 1,
+        Error = 2,
+        IncreaseMastery = 3,
+        LoadTiles = 4,
+        CatchItem = 5,
+        PrizeFish = 6,
+        LoadAlbum = 7,
+        CatchFish = 8,
+        Start = 9,
+        Auto = 10,
+        Simulate = 11,
+    }
+
+    public static ByteWriter Prepare(long rodUid) {
+        var pWriter = Packet.Of(SendOp.Fishing);
+        pWriter.Write<Command>(Command.Prepare);
+        pWriter.WriteLong(rodUid);
+
+        return pWriter;
+    }
+
+    public static ByteWriter Stop() {
+        var pWriter = Packet.Of(SendOp.Fishing);
+        pWriter.Write<Command>(Command.Stop);
+
+        return pWriter;
+    }
+
+    public static ByteWriter Error(FishingError error) {
+        var pWriter = Packet.Of(SendOp.Fishing);
+        pWriter.Write<Command>(Command.Error);
+        pWriter.Write<FishingError>(error);
+
+        return pWriter;
+    }
+
+    public static ByteWriter IncreaseMastery(int fishId, int exp, CaughtFishType fishType) {
+        var pWriter = Packet.Of(SendOp.Fishing);
+        pWriter.Write<Command>(Command.IncreaseMastery);
+        pWriter.WriteInt(fishId);
+        pWriter.WriteInt(exp);
+        pWriter.Write<CaughtFishType>(fishType);
+        pWriter.WriteShort(1); // fish grade/rarity ?
```

Comment:
it's not rarity, I think it's those like fishing tiers (beginner, intermediate, etc) maybe it does something special when you level up?